src/silx/gui/colors.py

```diff
@@ -556,22 +563,38 @@ def getGammaNormalizationParameter(self) -> float:
         """Returns the gamma correction parameter value."""
         return self.__gamma
 
-    def getAutoscaleMode(self) -> str:
-        """Return the autoscale mode of the colormap ('minmax' or 'stddev3')"""
+    def getAutoscaleMode(self) -> Literal["minmax", "stddev3", "percentile"]:
+        """Return the autoscale mode of the colormap."""
         return self._autoscaleMode
 
-    def setAutoscaleMode(self, mode: str):
-        """Set the autoscale mode: either 'minmax' or 'stddev3'
+    def setAutoscaleMode(self, mode: Literal["minmax", "stddev3", "percentile"]):
+        """Set the autoscale mode.
 
         :param mode: the mode to set
         """
         if self.isEditable() is False:
             raise NotEditableError("Colormap is not editable")
+        if mode == self.PERCENTILE_1_99:
+            deprecated_warning(
+                type_="Mode",
+                name="mode",
+                replacement="percentile",
+                since_version="3.0",
+            )
+            mode = self.PERCENTILE
+            self.setAutoscalePercentile((1.0, 99.0))
         assert mode in self.AUTOSCALE_MODES
         if mode != self._autoscaleMode:
             self._autoscaleMode = mode
             self.sigChanged.emit()
 
+    def setAutoscalePercentile(self, value: tuple[float, float]):
+        self._percentiles = value
+        self.sigChanged.emit()
+
+    def getAutoscalePercentile(self) -> tuple[float, float]:
```

Comment:
Worth a docstring IMO to explain that the 2-tuple is a (min, max) range with values expected in [0, 100]